cmd/gonstructor/gonstructor.go

```diff
@@ -20,29 +19,44 @@ const (
 	builderConstructorType = "builder"
 )
 
-var (
-	typeName          = flag.String("type", "", "[mandatory] a type name")
-	output            = flag.String("output", "", `[optional] output file name (default "srcdir/<type>_gen.go")`)
-	constructorTypes  = flag.String("constructorTypes", allArgsConstructorType, fmt.Sprintf(`[optional] comma-separated list of constructor types; it expects "%s" and "%s"`, allArgsConstructorType, builderConstructorType))
-	shouldShowVersion = flag.Bool("version", false, "[optional] show the version information")
-	withGetter        = flag.Bool("withGetter", false, "[optional] generate a constructor along with getter functions for each field")
-	initFunc          = flag.String("init", "", "[optional] name of function to call on object after creating it")
-)
+type stringArrayFlag []string
+
+func (a *stringArrayFlag) String() string {
+	return strings.Join(*a, ", ")
+}
+
+func (a *stringArrayFlag) Set(item string) error {
+	*a = append(*a, item)
+	return nil
+}
 
 func main() {
+	var typeNames stringArrayFlag
+	flag.Var(&typeNames, "type", `[mandatory] A type name. It accepts this option occurs multiple times to output the generated code of the multi types into a single file. If this option is given multiple times, the "-output" option becomes mandatory.`)
```

Comment:
@moznion maybe do `types=A,B,C,D,E,F`?

For backward compatibility, you can keep `type`, if u want.

Reply:
Yea, I intend to keep the backward compatibility. I think this way to specify the multiple options is suitable rather than comma-separated.
So if I have to have the consistency of the way to give this command's options, it should change `-constructorTypes` To `-constructorType` as well as this change.

Reply:
As u with :)